scipy/integrate/_cubature.py

```diff
@@ -0,0 +1,328 @@
+import heapq
+import itertools
+
+import numpy as np
+
+
+class CubatureRule:
+    def __init__(self, nodes, weights):
+        self.nodes = nodes
+        self.weights = weights
+
+    # returns the dimension of the cubature rule
+    # TODO: need to be more precise about what this means
+    def dimension(self):
+        return self.nodes.shape[-1]
+
+    # returns integral_estimate, error_estimate
+    def estimate(self):
+        pass
```

Comment:
Nit: if it's effectively an abstract class, signal this with a `raise NotImplementedError`